### PR TITLE
Update graal-sdk to 23.0.1 to enable Java 11 compatibility

### LINKED ISSUE #35054: Upgrade graal-sdk version to 23.0.1
### Description

`graal-sdk` 23.0.1 has just been released to Maven and it seems to include the fix the Java 11 compat:

```
$ javap -v ./x/com/oracle/svm/core/annotate/Substitute.class | grep major
  major version: 55
```

For comparison, 23.0.0 was only compatible with Java 17:

```
$ javap -v ./x/com/oracle/svm/core/annotate/Substitute.class | grep major
  major version: 61
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by galderz**
Closes #35054

### CODE REVIEW

File: independent-projects/bootstrap/pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
         <jboss-logmanager-embedded.version>1.0.11</jboss-logmanager-embedded.version>
         <slf4j-jboss-logmanager.version>1.1.0.Final</slf4j-jboss-logmanager.version>
         <slf4j-api.version>1.7.36</slf4j-api.version>
-        <graal-sdk.version>23.1.2</graal-sdk.version>
+        <graal-sdk.version>23.0.1</graal-sdk.version>
```
Comment (machi1990) [inline independent-projects/bootstrap/pom.xml:74]: The version here is being downgraded, is that intended?

Comment (gsmet) [reply independent-projects/bootstrap/pom.xml:74]: No the version was actually incorrect due to a bad search/replace by yours truly. So @galderz is fixing it in passing.

Comment (galderz) [reply independent-projects/bootstrap/pom.xml:74]: That version was wrong. A search/replace gone wrong https://github.com/quarkusio/quarkus/pull/35055

Comment (machi1990) [reply independent-projects/bootstrap/pom.xml:74]: Right, thanks.

**Review by zakkak — APPROVED** [review]

**Comment by galderz** [comment]
Added the backport tag so that it can be considered for backporting to 3.2.

**Pull request merged**